timber-lint/src/main/java/timber/lint/WrongTimberUsageDetector.java

```diff
@@ -560,6 +570,110 @@ private static boolean checkConditionalUsage(JavaContext context, UCallExpressio
     return checkElement(context, call, elseElement);
   }
 
+  private static LintFix quickFixIssueLog(UCallExpression logCall) {
+    List<UExpression> arguments = logCall.getValueArguments();
+    String methodName = logCall.getMethodName();
+    UExpression tag = arguments.get(0);
+
+    // 1st suggestion respects author's tag preference.
+    // 2nd suggestion drops it (Timber defaults to calling class name).
+    String fixSource1 = "Timber.tag(" + tag.asSourceString() + ").";
+    String fixSource2 = "Timber.";
+
+    int numArguments = arguments.size();
+    if (numArguments == 2) {
+      UExpression msgOrThrowable = arguments.get(1);
+      fixSource1 += methodName + "(" + msgOrThrowable.asSourceString() + ")";
+      fixSource2 += methodName + "(" + msgOrThrowable.asSourceString() + ")";
+    } else if (numArguments == 3) {
+      UExpression msg = arguments.get(1);
+      UExpression throwable = arguments.get(2);
+      fixSource1 +=
+          methodName + "(" + throwable.asSourceString() + ", " + msg.asSourceString() + ")";
+      fixSource2 +=
+          methodName + "(" + throwable.asSourceString() + ", " + msg.asSourceString() + ")";
+    } else {
+      throw new IllegalStateException("android.util.Log overloads should have 2 or 3 arguments");
+    }
+
+    String logCallSource = logCall.asSourceString();
+    LintFix.GroupBuilder fixGrouper = fix().group();
+    fixGrouper.add(
+        fix().replace().text(logCallSource).shortenNames().reformat(true).with(fixSource1).build());
+    fixGrouper.add(
+        fix().replace().text(logCallSource).shortenNames().reformat(true).with(fixSource2).build());
+    return fixGrouper.build();
+  }
+
+  private static LintFix quickFixIssueFormat(UCallExpression stringFormatCall) {
+    // Handles:
+    // 1) String.format(..)
+    // 2) format(...) [static import]
+    UExpression callReceiver = stringFormatCall.getReceiver();
+    String callSourceString = callReceiver == null ? "" : callReceiver.asSourceString() + ".";
+    callSourceString += stringFormatCall.getMethodName();
+
+    return fix().name("Remove String.format(...)").composite() //
+        // Delete closing parenthesis of String.format(...)
+        .add(fix().replace().pattern(callSourceString + "\\(.*(\\))").with("").build())
```

Comment:
I'm curious what this does to `String.format("%s%s", (1 + 2), 3)`

Reply:
Took that into account.  The .* is greedy.
![string-format](https://user-images.githubusercontent.com/1868149/30765918-fe7a0c14-9fbf-11e7-8d23-166a6b30a1be.gif)